newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -281,6 +284,38 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     return preferences;
   }
 
+  _getLastUsedPath(project: gdProject, kind: ResourceKind) {
+    const projectPath = path.dirname(project.getProjectFile());
+    const { values } = this.state;
+    const projectPaths = values.projectLastUsedPaths[projectPath];
+    if (projectPaths && projectPaths[kind]) {
+      return projectPaths[kind];
+    }
+    return '';
+  }
+
+  _setLastUsedPath(project: gdProject, kind: ResourceKind, latestPath: string) {
+    const projectPath = path.dirname(project.getProjectFile());
```

Comment:
path can be null (on the web-app, optionalRequire('path') will return null). Make your code robust against this in case this is called in the web-app (should not, but still):

`if (!path) return null;`

Reply:
Actually, do we want to store the project directory? No! :) 
There might be more than one project in a folder (unusual, but can happen).

So we want to save the filename to the project... in other words: 
```suggestion
    const projectPath = project.getProjectFile();
```